examples/cpu/seq_example.cpp

```diff
@@ -59,45 +59,57 @@ int seq_run(const std::string& detector_file, const std::string& cells_dir, unsi
-
-        for (std::size_t i = 0; i < cells_per_event.cells.size(); ++i )
+        traccc::host_measurement_container measurements_per_event;
+        traccc::host_spacepoint_container spacepoints_per_event;
+        measurements_per_event.headers.reserve(cells_per_event.headers.size());
+	measurements_per_event.items.reserve(cells_per_event.headers.size());
+        spacepoints_per_event.headers.reserve(cells_per_event.headers.size());
+	spacepoints_per_event.items.reserve(cells_per_event.headers.size()); 
+
+        for (std::size_t i = 0; i < cells_per_event.items.size(); ++i )
         {
+	    auto& module = cells_per_event.headers[i];
+	    module.pixel = traccc::pixel_segmentation{-8.425, -36.025, 0.05, 0.05};
+
             // The algorithmic code part: start
-            traccc::cluster_collection clusters_per_module = cc(cells_per_event.cells[i], cells_per_event.modules[i]);
-            clusters_per_module.position_from_cell = traccc::pixel_segmentation{-8.425, -36.025, 0.05, 0.05};
-            traccc::measurement_collection measurements_per_module = mt(clusters_per_module);
-            traccc::spacepoint_collection spacepoints_per_module = sp(measurements_per_module);
+            traccc::cluster_collection clusters_per_module = cc(cells_per_event.items[i], cells_per_event.headers[i]);
+            clusters_per_module.position_from_cell = module.pixel;
+
+            traccc::host_measurement_collection measurements_per_module = mt(clusters_per_module, module);
+            traccc::host_spacepoint_collection spacepoints_per_module = sp(module, measurements_per_module);
             // The algorithmnic code part: end
 
-            n_cells += cells_per_event.cells[i].size();
+            n_cells += cells_per_event.items[i].size();
             n_clusters += clusters_per_module.items.size();
-            n_measurements += measurements_per_module.items.size();
-            n_space_points += spacepoints_per_module.items.size();
-
-            measurements_per_event.push_back(std::move(measurements_per_module));
-            spacepoints_per_event.push_back(std::move(spacepoints_per_module));
+            n_measurements += measurements_per_module.size();
+            n_space_points += spacepoints_per_module.size();
+
+            measurements_per_event.items.push_back(std::move(measurements_per_module));
+	    measurements_per_event.headers.push_back(module);
+	    
+            spacepoints_per_event.items.push_back(std::move(spacepoints_per_module));
+	    spacepoints_per_event.headers.push_back(module.module);
         }
 
         traccc::measurement_writer mwriter{std::string("event")+event_number+"-measurements.csv"};
-        for (const auto& measurements_per_module : measurements_per_event){
-            auto module = measurements_per_module.module;
-            for (const auto& measurement : measurements_per_module.items){
+	for (int i=0; i<measurements_per_event.items.size(); ++i){
```

Comment:
You should change `i` into an `std::size_t` variable. As the build currently emits the following warning:

```
/data/ssd-1tb/projects/traccc/traccc/examples/cpu/seq_example.cpp:98:17: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<std::vector<traccc::measurement, std::pmr::polymorphic_allocator<traccc::measurement> >, std::pmr::polymorphic_allocator<std::vector<traccc::measurement, std::pmr::polymorphic_allocator<traccc::measurement> > > >::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
   98 |  for (int i=0; i<measurements_per_event.items.size(); ++i){
      |                ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

(The "most exact" thing would be to make use of the `size_type` definition of the type that you're using. But let's not get crazy here. `std::size_t` will work fine.)

Reply:
good comment. changed to `size_t` for now